src/transformers/pipelines/text2text_generation.py

```diff
@@ -96,7 +96,7 @@ def __call__(
             )
 
         with self.device_placement():
-            inputs = self._parse_and_tokenize(*documents, padding=padding)
+            inputs = self._parse_and_tokenize(*documents, padding=padding, **generate_kwargs)
```

Comment:
I think I'd prefer to either pop all tokenizer relevant tokens from `generate_kwargs` before this statement and only insert those or even directly add them as `keyword` arguments to the `__call__` method, like `padding_strategy=None` e.g.

Reply:
I like that too, but that breaks backward compatibility right ? (If there was code written with an ignored kwarg, then we would start failing).

I could start something like: 

```python
__call__(self, tokenizer_arg1=XXX, tokenizer_arg2=XXXX, generate_kwarg1=XXX, ...., **kwargs):
  if kwargs:
      warnings.warn(UserWarning, "Unused keyword arguments {kwargs}" for pipeline {self}")

   self._parse_and_tokenizer(tokenizer_arg1=tokenizer_arg1, tokenizer_arg2=tokenizer_arg2)

   self.model.generate(generate_arg1, generate_kwarg2=generate_kwarg2, ...)
```
The only drawback of this, is we're calling `self.model.generate` over which I don't know if we control all kwargs. I mean are there model specific kwarg that we want to pass too (so we can't have an exhaustive list within `__call__` because we don't know ahead of time which arguments are available ?

Reply:
I don't understand why this would break backward compatibility. Previously no kwars were passed to `_parse_and_tokenize` => so the user had no control over the tokenizer at the `__call__` method of the pipeline. Now if we add kwargs, there should not be a problem in adding all possible tokenizer kwargs when correctly setting their defaults to the defaults in tokenizer no? But I see that there are a lot of tokenizer kwargs which don't all default to `None` :-/ Hmm, maybe the best solution is to just allow the `truncate` tokenizer argument for now and not the rest?

In general, as discussed with @LysandreJik and @mfuntowicz already multiple times, IMO the current pipelines design in Transformers is just not general enough to cleanly allow all use cases...

I think as it is now is quite dangerous if for some reason generate kwargs get the same name as tokenizer kwargs. 

It's definitely not an option to do `generate_kwags1` .... => there are way to many kwargs and maintainability becomes an issue?

Reply:
Actually, I just noticed that kwargs is not even really used in `_parse_and_tokenize` => then I think there is no problem in popping just `add_special_token` and `truncation` and not passing all the `generation_kwargs` -> why would this break anything?

Reply:
I changed that to follow your recommendation. Much better this way ! Thanks.